index.go

```diff
@@ -156,42 +157,27 @@ func (idx *faissIndex) Search(x []float32, k int64) (
 	return
 }
 
-func (idx *faissIndex) SearchWithoutIDs(x []float32, k int64, exclude []int64) (
+func (idx *faissIndex) SearchWithoutIDs(x []float32, k int64, exclude []int64, params json.RawMessage) (
 	distances []float32, labels []int64, err error,
 ) {
-	if len(exclude) <= 0 {
-		return idx.Search(x, k)
+	var selector *C.FaissIDSelector
+	if len(exclude) > 0 {
+		excludeSelector, err := NewIDSelectorNot(exclude)
+		if err != nil {
+			return nil, nil, err
+		}
+		selector = excludeSelector.sel
+		defer excludeSelector.Delete()
 	}
 
-	excludeSelector, err := NewIDSelectorNot(exclude)
+	searchParams, err := NewSearchParamsIVF(idx, params, selector)
```

Comment:
Let's do this if and only params is non nil.

Reply:
Same comment as below